src/d3d9/d3d9_swapchain.cpp

```diff
@@ -9,6 +9,43 @@
 
 namespace dxvk {
 
+
+  struct D3D9WindowData {
+    bool unicode;
+    WNDPROC proc;
+  };
+
+
+  static std::recursive_mutex windowProcMapMutex;
+  static std::unordered_map<HWND, D3D9WindowData> windowProcMap;
+
+
+  static D3D9WindowData GetD3D9WindowData(HWND window) {
+    std::lock_guard<std::recursive_mutex> lock(windowProcMapMutex);
+
+    auto it = windowProcMap.find(window);
+    if (it == windowProcMap.end())
+      return {};
+
+    return it->second;
+  }
+
+
+  static LRESULT CALLBACK D3D9WindowProc(HWND window, UINT message, WPARAM wparam, LPARAM lparam)
+  {
+      if (message == WM_NCCALCSIZE && wparam == TRUE)
+        return 0;
+
+      auto windowData = GetD3D9WindowData(window);
+      if (!windowData.proc)
+        return 0;
```

Comment:
``return DefWindowProcA(...)``?

Reply:
ofc

Reply:
Although we have lost the unicode information as well...

Reply:
I used IsWindowUnicode here as well, I'm not sure why it was avoided in wined3d, maybe it's expensive. In any case this case should hopefully not happen.